frame/timestamp/src/lib.rs

```diff
@@ -151,15 +151,22 @@ decl_module! {
 		/// 
 		/// # <weight>
 		/// - `O(T)` where `T` complexity of `on_timestamp_set`
-		/// - 2 storage mutations (codec `O(1)`).
+		/// - 2 storage mutations (codec `O(1)`). (effectively 1, because of `DidUpdate::take` in `on_finalize`)
```

Comment:
I guess DidUpdate will still be searched in the trie, I think it should count for one read and zero write.

Reply:
But I thought that DB access is when it goes to the database and that doesn't happen here, right?

Reply:
AFAIK a read is just a read from database, even if you write nothing you still have read from database. @shawntabrizi

Reply:
I agree that the total db operations of `set` should be:

* 1 read: DidUpdate::exists()
* 1 write: Now::put(now)

We should ignore `DidUpdate::put(true)` because it is always removed at the end of the block, and thus never written to the underlying database.

Reply:
we don't count `Self::now()` as a read ? it looks to be 2 read and 1 write to me

Reply:
Yes, sorry I missed it (i dont know how??)

then 2 read, 1 write